File: miniserver/src/main/java/de/miniserver/ServerConfig.java
package de.miniserver;

import de.mein.konsole.KResult;
import de.mein.sql.Pair;

import java.util.HashMap;
import java.util.Map;

public class ServerConfig implements KResult {
    private String certPath;
    private String workingDirectory;
    private String pubKeyPath;
    private String certName;

    public String getPubKeyPath() {
        return pubKeyPath;
    }

    public void setPubKeyPath(String pubKeyPath) {
        this.pubKeyPath = pubKeyPath;
    }

    public String getPrivKeyPath() {
        return privKeyPath;
    }

    public void setPrivKeyPath(String privKeyPath) {
        this.privKeyPath = privKeyPath;
    }

    private String privKeyPath;
    private Map<String, Pair<String>> files = new HashMap<>();

    public String getCertPath() {
        return certPath;
    }

    public void setCertPath(String certPath) {
        this.certPath = certPath;
    }

    public Map<String, Pair<String>> getFiles() {
        return files;
    }

    public ServerConfig addEntry(String binaryFile, String name, String versionFile) {
        files.put(binaryFile, new Pair<>(String.class, name, versionFile));
        return this;
    }

    public void setWorkingDirectory(String workingDirectory) {
        this.workingDirectory = workingDirectory;
    }

    public String getWorkingDirectory() {
        return workingDirectory;
    }

    public void setCertName(String certName) {
        this.certName = certName;
    }

    public String getCertName() {
        return certName;
    }
}

File: miniserver/src/main/kotlin/de/miniserver/ServerConfig.kt
package de.miniserver

import de.mein.KResult
import de.mein.auth.data.MeinAuthSettings
import de.mein.sql.Pair
import java.io.File
import java.util.*

class ServerConfig : KResult {
    var certPath: String? = null
    var workingPath: String? = DEFAULT_WORKING_DIR.absolutePath
    var pubKeyPath: String? = null
    var certName: String? = null

    var privKeyPath: String? = null
    private val files = HashMap<String, Pair<String>>()
    var httpPort: Int? = null
    var httpsPort: Int? = null
    var authPort: Int = DEFAULT_AUTH
    var transferPort: Int = DEFAULT_TRANSFER
    var pipes: Boolean = true
    var workingDirectory: File? = null
        get() = File(workingPath)
    var keySize: Int = 2048
    var restartCommand = mutableListOf<String>()
    var keepBinaries: Boolean = false
    var preserveLogLinesInDb = 2000L


    fun getFiles(): Map<String, Pair<String>> {
        return files
    }

    fun addEntry(binaryFile: String, name: String, versionFile: String): ServerConfig {
        files[binaryFile] = Pair(String::class.java, name, versionFile)
        return this
    }

    companion object {
        val DEFAULT_WORKING_DIR = File("server")
        val DEFAULT_AUTH: Int = MeinAuthSettings.UPDATE_MSG_PORT
        const val DEFAULT_TRANSFER: Int = MeinAuthSettings.UPDATE_BINARY_PORT
        const val DEFAULT_HTTP: Int = 8080
        const val DEFAULT_HTTPS: Int = 8443
    }
}
